Title bar and stored examples

inaimathi opened the pull request.

inaimathi committed:
Start on title bar

inaimathi committed:
Initial example files

inaimathi committed:
Add default stored example

- Add default stored example with README from the former
  help-message (which didn't seem to appear in the help pane for some
  reason)
- add About pane to right hand side
- change current example based on URL hash element

inaimathi committed:
Add example dropdowns to the menu

inaimathi committed:
Compute example links from available example directories

inaimathi committed:
Add hack to get the toolbar working as expected

inaimathi committed:
Minor renaming and use the :read-only option on editor!

- Use the :read-only option on editor! instead of manually calling
  .setOption on the results
- Rename load-example-from-hash! to load-example-from-page-hash!

inaimathi committed:
Some styling finesse

jamesaoverton commented:
Looking good. A few things:

Interface:
- The "Select a View" button is nice. I would like to see the name of the current view without clicking the button.

Bugs (Firefox and Safari):
- The Examples dropdown does not change the Content or Context. Firefox says "XML Parsing Error: not well-formed Location: https://fiddle.knotation.org.ontodev.com/static/example/D/README.md Line Number 1, Column 2:"
- Context is an empty CodeMirror (I see line number 1 with no content) until I click inside it. ![context](https://user-images.githubusercontent.com/292409/36155898-97779d6a-10a3-11e8-8038-bb2dedaa5d21.jpeg)
- The bottom margin of the "Fork me on GitHub" ribbon is much smaller than the top margin. ![ribbon](https://user-images.githubusercontent.com/292409/36155834-6e737c36-10a3-11e8-923e-220ec8010464.jpeg)

inaimathi committed:
Properly refresh editors in apply-hacks!

inaimathi committed:
Fix Content-Type error thing

- Fix front-end XML parsing errors in Firefox (root cause was lack of
  Content Type header with .kn files from http-kit. Solved by adding a
  separate handler to return the file contents for a given example all
  together in JSON format)
- Fix setValue errors (error gets tripped somewhere in CodeMirror hook
  handling after the value is set. This may be an issue with the
  parser doing odd things with empty editors. For the moment, ignoring
  errors on those setValue calls)

inaimathi committed:
Fix alignment issue with github fork ribbon

inaimathi committed:
Use local github-fork-ribbon CSS rather than CDN

inaimathi commented:
That should address all three comments

inaimathi committed:
Update button dropdown text on selection
